src/components/header/modals/ModalPolkasafe.vue

```diff
@@ -0,0 +1,394 @@
+    onHeightChange();
+
+    const displayBalance = (balance: string): number => {
+      return truncate(ethers.utils.formatEther(balance || '0'));
+    };
+
+    const handleGetMultisigAccounts = async (): Promise<void> => {
+      multisigAccounts.value = [];
+      isLoadingPolkasafe.value = true;
+      selAccount.value = '';
+      if (!selectedSignatory.value || currentNetworkChain.value !== astarChain.ASTAR) {
+        return;
+      }
+      try {
+        const extensions = await web3Enable('AstarNetwork/astar-apps');
+        const injector = extensions.find((it) => {
+          return selectedSignatory.value && it.name === selectedSignatory.value.source;
+        }) as any;
+
+        if (!injector) {
+          throw Error('injector not found');
+        }
+        const substratePrefix = 42;
+        const signatory = encodeAddress(selectedSignatory.value.address, substratePrefix);
+
+        const setMultisigAccounts = async (c: Polkasafe): Promise<void> => {
+          const { data, error } = await c.connectAddress(signatory);
+          if (error) throw Error(error);
+          if (!data.hasOwnProperty('multisigAddresses')) {
+            throw Error('error fetching multisig accounts');
+          }
+
+          //@ts-ignore
+          const filteredMultisigAccounts = data.multisigAddresses.filter((it: MultisigAddress) =>
+            isValidAddressPolkadotAddress(it.address, ASTAR_ADDRESS_PREFIX)
+          );
+          if (filteredMultisigAccounts.length > 0) {
+            const updatedAccountMap = await Promise.all(
+              filteredMultisigAccounts.map(async (it: MultisigAddress) => {
+                const balance = await fetchNativeBalance({
+                  api: $api as ApiPromise,
+                  address: it.address,
+                });
+                return { ...it, balance };
+              }) as MultisigAddress[]
+            );
+            multisigAccounts.value = updatedAccountMap.sort(
+              (a: MultisigAddress, b: MultisigAddress) => Number(b.balance) - Number(a.balance)
+            );
+          }
+        };
+
+        const handleInitializePolkasafe = async (): Promise<void> => {
+          const client = new Polkasafe();
+          await client.connect('astar', signatory, injector);
+          container.addConstant<Polkasafe>(Symbols.PolkasafeClient, client);
+          await setMultisigAccounts(client);
+        };
+
+        try {
+          const polkasafeClient = container.get<Polkasafe>(Symbols.PolkasafeClient);
+          const isSigned = polkasafeClient && polkasafeClient.address === signatory;
+          if (isSigned) {
+            await setMultisigAccounts(polkasafeClient);
+          } else {
+            await handleInitializePolkasafe();
+          }
+        } catch (error) {
+          // Memo: polkasafeClient has not been initialized
+          await handleInitializePolkasafe();
+        }
+      } catch (error) {
+        console.error(error);
```

Comment:
Don't swallow, handle in line 346 and show error toast

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/834/commits/6d8f31f2645b3e46060408ae8ca42e1fa7ba542d) commit.